MFVNeutralino/plugins/Vertexer.cc

```diff
@@ -1,891 +1,2008 @@
+          for (nv[1] = nv[0] + 1; nv[1] != vertices->end(); ++nv[1]) {
+
+            Measurement1D nv_dist = vertex_dist(*nv[0], *nv[1]);
+            printf("  new vertex dist (2d? %i) %7.3f  sig %7.3f\n", use_2d_vertex_dist, nv_dist.value(), nv_dist.significance());
+          }
+        }
+      }
+    }
+  }
+
+  //////////////////////////////////////////////////////////////////////
+  // Drop tracks that "move" the vertex too much by refitting without each track.
+  //////////////////////////////////////////////////////////////////////
+
+  if (max_nm1_refit_dist3 > 0 || max_nm1_refit_distz > 0) {
+    std::vector<int> refit_count(vertices->size(), 0);
+
+    int iv = 0;
+    for (v[0] = vertices->begin(); v[0] != vertices->end(); ++v[0], ++iv) {
+      if (max_nm1_refit_count > 0 && refit_count[iv] >= max_nm1_refit_count)
+        continue;
+
+      const track_vec tks = vertex_track_vec(*v[0]);
+      const size_t ntks = tks.size();
+      if (ntks < 3)
+        continue;
+
+      if (verbose) {
+        printf("doing n-%i refit on vertex at %7.4f %7.4f %7.4f with %lu tracks\n", refit_count[iv] + 1, v[0]->x(), v[0]->y(), v[0]->z(), ntks);
+        for (size_t i = 0; i < ntks; ++i)
+          printf("  refit %lu will drop tk pt %7.4f +- %7.4f eta %7.4f +- %7.4f phi %7.4f +- %7.4f dxy %7.4f +- %7.4f dz %7.4f +- %7.4f\n", i, tks[i]->pt(), tks[i]->ptError(), tks[i]->eta(), tks[i]->etaError(), tks[i]->phi(), tks[i]->phiError(), tks[i]->dxy(), tks[i]->dxyError(), tks[i]->dz(), tks[i]->dzError());
+      }
+
+      std::vector<reco::TransientTrack> ttks(ntks - 1);
+      for (size_t i = 0; i < ntks; ++i) {
+        for (size_t j = 0; j < ntks; ++j)
+          if (j != i)
+            ttks[j - (j >= i)] = tt_builder->build(tks[j]);
+
+        reco::Vertex vnm1(TransientVertex(kv_reco->vertex(ttks)));
+        const double dist3_2 = mag2(vnm1.x() - v[0]->x(), vnm1.y() - v[0]->y(), vnm1.z() - v[0]->z());
+        const double distz = mag(vnm1.z() - v[0]->z());
+        if (verbose) printf("  refit %lu chi2 %7.4f vtx %7.4f %7.4f %7.4f dist3 %7.4f distz %7.4f\n", i, vnm1.chi2(), vnm1.x(), vnm1.y(), vnm1.z(), sqrt(dist3_2), distz);
+
+        if (vnm1.chi2() < 0 ||
+            (max_nm1_refit_dist3 > 0 && mag2(vnm1.x() - v[0]->x(), vnm1.y() - v[0]->y(), vnm1.z() - v[0]->z()) > pow(max_nm1_refit_dist3, 2)) ||
+            (max_nm1_refit_distz > 0 && distz > max_nm1_refit_distz)) {
+          if (verbose) {
+            printf("    replacing");
+            if (refit_count[iv] < max_nm1_refit_count - 1)
+              printf(" and reconsidering");
+            printf("\n");
+          }
+
+          *v[0] = vnm1;
+          ++refit_count[iv];
+          --v[0], --iv;
+          break;
+        }
+      }
+    }
+  }
+
+
+  // FIXME JPR continue review from here! 
+  if (histos_output_afterdzfit){
+    std::map<reco::TrackRef, int> track_use;
+    int count_5trk_vertices = 0;
+
+    typedef std::vector<reco::TrackRef> track_vec;

```

Comment:
This is already typedef-ed in the class definition, it doesn't do anything here, right?

Reply:
yes, just remove it. Done.